src/data/nat/basic.lean

```diff
@@ -286,6 +286,10 @@ lemma _root_.has_lt.lt.nat_succ_le {n m : ℕ} (h : n < m) : succ n ≤ m := suc
 lemma succ_eq_one_add (n : ℕ) : n.succ = 1 + n :=
 by rw [nat.succ_eq_add_one, nat.add_comm]
 
+lemma succ_sub_self : ∀ {n : ℕ}, n.succ - n = 1
```

Comment:
I think this is just `add_tsub_cancel_left`; is that imported here?

Reply:
Eh yeah, it is. I thought it would be interesting to have this syntactical variation but on a second thought, meh.